fili-core/src/main/java/com/yahoo/bard/webservice/metadata/LookupLoadTask.java

```diff
@@ -0,0 +1,124 @@
+    public static final String INITIAL_LOOKUP_CHECKING_DELAY = SYSTEM_CONFIG.getPackageVariableName(
+            "initial_lookup_checking_delay"
+    );
+
+    private final DruidWebService druidClient;
+    private final DimensionDictionary dimensionDictionary;
+    private final SuccessCallback successCallback;
+    private final FailureCallback failureCallback;
+    private final HttpErrorCallback errorCallback;
+    private Set<String> unloadedLookups;
+
+    /**
+     * Constructor.
+     *
+     * @param druidClient  The client to query Druid coordinator
+     * @param dimensionDictionary  A {@link com.yahoo.bard.webservice.data.dimension.DimensionDictionary} that is used
+     * to obtain a list of lookups in Fili.
+     */
+    public LookupLoadTask(DruidWebService druidClient, DimensionDictionary dimensionDictionary) {
+        super(
+                LookupLoadTask.class.getSimpleName(),
+                SYSTEM_CONFIG.getLongProperty(INITIAL_LOOKUP_CHECKING_DELAY, 0),
+                SYSTEM_CONFIG.getLongProperty(LOOKUP_NORMAL_CHECKING_PERIOD_KEY, TimeUnit.MINUTES.toMillis(1))
+        );
+        this.druidClient = druidClient;
+        this.dimensionDictionary = dimensionDictionary;
+        this.successCallback = buildLookupSuccessCallback();
+        this.failureCallback = getFailureCallback();
+        this.errorCallback = getErrorCallback();
+    }
+
+    @Override
+    public void run() {
+        // download load statuses of all lookups
+        druidClient.getJsonObject(successCallback, errorCallback, failureCallback, LOOKUP_QUERY);
+    }
+
+    /**
+     * Returns a set of lookup namespaces that have not been loaded to Druid yet.
+     *
+     * @return the set of lookup namespaces that have not been loaded to Druid yet
+     */
+    public Set<String> getUnloadedLookups() {
+        return unloadedLookups;
+    }
+
+    /**
+     * Returns a callback that has actions on lookups from a successful Druid response.
+     * <p>
+     * The callback obtains a complete list of configured lookups from Druid coordinator, compares this list against
+     * the list of lookups configured in Fili, and finds all lookup namespace names that are either not loaded yet in
+     * Druid or does not exist in Druid at all. These namespaces can be retrieved later by calling
+     * {@link #getUnloadedLookups()}.
+     *
+     * @return the callback that has actions on lookups from a successful Druid response
+     */
+    protected SuccessCallback buildLookupSuccessCallback() {
+        return rootNode -> {
+            Map<String, Boolean> lookupStatuses = new HashMap<>();
+            Iterator<Map.Entry<String, JsonNode>> entries = rootNode.fields();
+            while (entries.hasNext()) {
+                Map.Entry<String, JsonNode> entry = entries.next();
+                lookupStatuses.put(entry.getKey(), entry.getValue().get("loaded").asBoolean());
+            }
+
+            unloadedLookups = dimensionDictionary.findAll().stream()
```

Comment:
There are three states on lookups, loaded, toLoad and toDrop.  I think 'unloaded' is confusing because we usually mean toLoad but it sounds like they're being dropped.  how about 'pending' where you have 'unloaded'